### PR TITLE
Bug 1879640 - Suggest: Update Yelp prefix matching

### PR DESCRIPTION
**Pull request by 0c0w3**
### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by 0c0w3** [comment]
The only substantive change is to lower the subject prefix threshold from 6 to 2. I talked with Loren and we came up with these rules for prefix matching:

* The entire search string has to be at least 5 characters long, **and**
* At least 2 characters of the subject have to be typed

The 5 char overall limit needs to be configurable. We've added a Nimbus variable for it and desktop will enforce it. Maybe we can also add some Yelp RS config and have the Rust component enforce it, but as long as we have a Nimbus variable, the client will need to do it.

I renamed `top_pick` back to @dadaa's original `subject_exact_match`. Desktop will use this to bypass the 5 char limit for subjects that are shorter than 5 chars. `top_pick` doesn't fit this use case and is too prescriptive.

File: components/suggest/src/yelp.rs [hunk 1]
```diff
@@ -360,7 +360,11 @@ impl<'a> SuggestDao<'a> {
         // Otherwise, apply prefix-match.
         Ok(
             match self.conn.query_row_and_then_cachable(
-                "SELECT keyword FROM yelp_subjects WHERE keyword BETWEEN :candidate AND :candidate || x'FFFF' ORDER BY LENGTH(keyword) ASC LIMIT 1",
+                "SELECT keyword
+                 FROM yelp_subjects
+                 WHERE keyword BETWEEN :candidate AND :candidate || x'FFFF'
+                 ORDER BY LENGTH(keyword) ASC, keyword ASC
```
Comment (0c0w3) [inline components/suggest/src/yelp.rs:366]: The only substantive change here is to add a backstop `keyword ASC` to the `ORDER BY` to ensure a consistent ordering for keywords that are the same length.

**Review by dadaa — APPROVED** [review]
Thanks!

**Review by ncloudioj — APPROVED** [review]
LGTM!

**Pull request merged**